cuda_core/cuda/core/experimental/_program.py

```diff
@@ -513,6 +607,31 @@ def compile(self, target_type, name_expressions=(), logs=None):
 
             return ObjectCode._init(data, target_type, symbol_mapping=symbol_mapping, name=self._options.name)
 
+        elif self._backend == "NVVM":
+            if target_type not in ("ptx", "ltoir"):
+                raise ValueError(f'NVVM backend only supports target_type="ptx", "ltoir", got "{target_type}"')
+
+            nvvm_options = self._translate_program_options_to_nvvm(self._options)
+            if target_type == "ltoir" and "-gen-lto" not in nvvm_options:
+                nvvm_options.append("-gen-lto")
+            nvvm = _get_nvvm_module()
+            nvvm.verify_program(self._mnff.handle, len(nvvm_options), nvvm_options)
```

Comment:
Given how hard it is to debug IR issues, we should wrap `verify` and `compile` in `_exception_manager` from `_linker.py` (copy/paste it to this file and modify it for now, I'll do a refactor later), like this
```python
with _exception_manager(self):
    nvvm.verify_program(...)
    nvvm.compile_program(...)
```
In the local `_exception_manager`, make sure the program log is fetched if any exception is raised.

Reply:
(We did not need `_exception_manager` for NVRTC because `handle_return` has a special path for it.)

Reply:
Ack, modified. refactor can be followed up , thanks